fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/audit/BulkAuditLogger.java

```diff
@@ -0,0 +1,311 @@
+     * @param responseStatus
+     *            The response status.
+     * @param location
+     *            the destination or source for the export or import
+     * @param users
+     *            the principals that initiated the request
+     * @throws Exception
+     */
+    public void logCreateOnImport(Resource newResource, Date startTime, Date endTime, Response.Status responseStatus, String location, String users)
+        throws Exception {
+        final String METHODNAME = "logCreateOnImport";
+        log.entering(CLASSNAME, METHODNAME);
+        if (shouldLog()) {
+            log(AuditLogEventType.FHIR_CREATE, "C", "FHIR BulkData Create request", null, newResource, startTime, endTime, responseStatus, null, null, location, users);
+        }
+        log.exiting(CLASSNAME, METHODNAME);
+    }
+
+    /**
+     * Builds an audit log entry for a 'validate' in a bulkdata service invocation.
+     *
+     * @param newResource
+     *            The Resource object being created.
+     * @param startTime
+     *            The start time of the create request execution.
+     * @param endTime
+     *            The end time of the create request execution.
+     * @param responseStatus
+     *            The response status.
+     * @param location
+     *            the destination or source for the export or import
+     * @param users
+     *            the principals that initiated the request
+     * @throws Exception
+     */
+    public void logValidateOnImport(Resource newResource, Date startTime, Date endTime, Response.Status responseStatus, String location, String users)
+        throws Exception {
+        final String METHODNAME = "logValidateOnImport";
+        log.entering(CLASSNAME, METHODNAME);
+        if (shouldLog()) {
+            log(AuditLogEventType.FHIR_OPERATION, "R", "FHIR BulkData Validate on Import request", null, newResource, startTime, endTime, responseStatus, null, null, location, users);
+        }
+        log.exiting(CLASSNAME, METHODNAME);
+    }
+
+    /**
+     * Builds an audit log entry for an 'update' in a bulkdata service invocation.
+     *
+     * @param oldResource
+     *            The previous version of the Resource, before it was updated.
+     * @param updatedResource
+     *            The updated version of the Resource.
+     * @param startTime
+     *            The start time of the update request execution.
+     * @param endTime
+     *            The end time of the update request execution.
+     * @param responseStatus
+     *            The response status.
+     * @param location
+     *            the destination or source for the export or import
+     * @param users
+     *            the principals that initiated the request
+     * @throws Exception
+     */
+    public void logUpdateOnImport(Resource oldResource, Resource updatedResource, Date startTime, Date endTime, Response.Status responseStatus, String location,
+        String users)
+        throws Exception {
```

Comment:
Applies to a number of long method declarations:  please use double-indent to offset from the first expression of the method